File: LeetCode/Easy/0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    private const int MaxNodes = 100;
    public IList<int> InorderTraversal(TreeNode root) {
        var values = new List<int>(MaxNodes);
        var stack = new Stack<TreeNode>(MaxNodes / 10);
        var current = root;

        while (current != null || stack.Count != 0)
        {
            if (current != null)
            {
                stack.Push(current);
                current = current.left;
            }
            else
            {
                current = stack.Pop();
                values.Add(current.val);
                current = current.right;
            }
        }

        return values;
    }
}

File: LeetCode/Easy/0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        current = root
        stack = []
        output = []

        while current or stack:
            while current:
                stack.append(current)
                current = current.left
            
            node = stack.pop()
            output.append(node.val)
            if node.right is not None:
                current = node.right
            
        return output
        


        